### PR TITLE
Some initial pins and feedstocks for s390x

### PR DESCRIPTION
**Pull request by npanpaliya**
## Checklist before submitting

- [ ] Did you read the [contributor guide](https://github.com/open-ce/open-ce/blob/main/CONTRIBUTING.md)?
- [ ] Did you update any affected [documentation](https://github.com/open-ce/open-ce/blob/main/doc/)?
- [ ] Did you write any tests to validate this change?

## Description

Fixes # (issue).

## Review process to land 

1. All tests and other checks must succeed.
2. At least one [maintainer](https://github.com/open-ce/open-ce/blob/main/MAINTAINERS.md) must review and approve.
3. If any  [maintainer](https://github.com/open-ce/open-ce/blob/main/MAINTAINERS.md) requests changes, they must be addressed.

### CODE REVIEW

File: envs/conda_build_config.yaml [hunk 1]
```diff
@@ -230,9 +233,11 @@ pybind11:
 pydantic:
   - 1.8.2
 pytest:
-  - 5.*
+  - 5.*   # [not s390x]
+  - 6.*   # [s390x]
 python:
-  - 3.8
+  - 3.7   # [not s390x]
```
Comment (cdeepali) [inline envs/conda_build_config.yaml:239]: I think we can remove 3.7.

File: envs/conda_build_config.yaml [hunk 2]
```diff
@@ -244,7 +248,8 @@ pytorch_scatter:
 pytorch_sparse:
   - 0.6.10
 pyyaml:
-  - 5.*
+  - 5.*   # [not s390x]
+  - 6.*   # [s390x]
```
Comment (ravigumm) [inline envs/conda_build_config.yaml:252]: pyyaml: Can we move back to 5.x on s390x ? We faced some conflicts in CP4D when we moved to tensorflow 2.7.x and we changed pyyaml back to 5.x

File: envs/conda_build_config.yaml [hunk 3]
```diff
@@ -282,6 +287,8 @@ srsly:
   - 2.*
 starlette:
   - 0.19.*
+sqlite:
+  - 3.38.2   # [s390x]
```
Comment (ravigumm) [inline envs/conda_build_config.yaml:291]: Do we want this to be exact version ? OR 3.* or 3.38.* is better ?

**Comment by ravigumm** [comment]
Other than the pyyaml version, LGTM. Thanks Nishidha.

File: envs/conda_build_config.yaml (hunk 2 above)
Comment (npanpaliya) [reply envs/conda_build_config.yaml:252]: This change is cherry-picked from 1.5 branch as it is https://github.com/open-ce/open-ce/blob/open-ce-r1.5/envs/conda_build_config.yaml#L240.

File: envs/conda_build_config.yaml (hunk 3 above)
Comment (npanpaliya) [reply envs/conda_build_config.yaml:291]: https://github.com/open-ce/open-ce/blob/open-ce-r1.5/envs/conda_build_config.yaml#L276

Comment (npanpaliya) [reply envs/conda_build_config.yaml:291]: I think we can relax it a bit by making 3.38.*.

**Review by cdeepali — APPROVED** [review]

File: envs/conda_build_config.yaml (hunk 2 above)
Comment (ravigumm) [reply envs/conda_build_config.yaml]: We were at 6.x level for some time. But faced issues because of some other packages and moved back to 5.x

**Pull request merged**